dynatrace-operator/chart/default/tests/Common/csi/daemonset_test.yaml

```diff
@@ -0,0 +1,41 @@
+suite: test deployment for csi DaemonSet
+templates:
+  - Common/csi/daemonset.yaml
+tests:
+
+  - it: should exist in case of cloudNativeFullStack mode
+    set:
+      operator.image: image-name
+      cloudNativeFullStack.enabled: true
+    asserts:
+      - isKind:
+          of: DaemonSet
+      - equal:
+          path: metadata.name
+          value: dynatrace-oneagent-csi-driver
+      - equal:
+          path: metadata.namespace
+          value: NAMESPACE
+      - isNotEmpty:
+          path: metadata.labels
+      - isNotEmpty:
+          path: spec.template.metadata.labels
+
+  - it: should exist in case of applicationMonitoring mode with useCSIDriver enabled
+    set:
+      operator.image: image-name
+      applicationMonitoring.enabled: true
+      applicationMonitoring.useCSIDriver: true
+    asserts:
+      - isKind:
+          of: DaemonSet
+      - equal:
+          path: metadata.name
+          value: dynatrace-oneagent-csi-driver
+      - equal:
+          path: metadata.namespace
+          value: NAMESPACE
+      - isNotEmpty:
+          path: metadata.labels
+      - isNotEmpty:
+          path: spec.template.metadata.labels
```

Comment:
A test checking if it does not exist in other cases is missing

Reply:
wanted to but wasn't able to do that, and we don't have any similar tests :/

Reply:
I take it back @chrismuellner found a solution :D